src/python/pants/source/filespec.py

```diff
@@ -11,7 +11,7 @@
 def globs_matches(
   paths: Iterable[str], patterns: Iterable[str], exclude_patterns: Iterable[str],
 ) -> bool:
-  path_globs = PathGlobs(include=patterns, exclude=exclude_patterns)
+  path_globs = PathGlobs(globs=(*patterns, *(f"!{e}" for e in exclude_patterns)))
```

Comment:
I personally like this syntax a lot!

Reply:
I didn't like having to preprocess the excludes like this at first, but it's growing on me.